src/js/model/storage.ts

```diff
@@ -1,30 +1,38 @@
 import { serialize } from 'utils/parser';
 import ApiSettings from 'api/api-settings';
+import type { GenericObject } from 'types/generic.type';
+import type { MediaModel } from 'controller/model';
 
 let storage = {
-    removeItem: function() {}
+    removeItem: function(itemName: string) {
+        // No unused arguments.
+        itemName;
+    }
 };
 
 try {
     storage = window.localStorage || storage;
 } catch (e) {/* ignore */}
 
-function Storage(namespace, persistItems) {
-    this.namespace = namespace;
-    this.items = persistItems;
-}
+class Storage {
+    namespace: string;
+    items: any[];
+    constructor(namespace: string, persistItems: GenericObject[]) {
+        this.namespace = namespace;
+        this.items = persistItems;
+    }
 
-Object.assign(Storage.prototype, {
-    getAllItems() {
+    getAllItems(): GenericObject {
         return this.items.reduce((memo, key) => {
             const val = storage[`${this.namespace}.${key}`];
             if (val) {
                 memo[key] = key !== 'captions' ? serialize(val) : JSON.parse(val);
             }
             return memo;
         }, {});
-    },
-    track(model) {
+    }
+
+    track(model: MediaModel): void {
```

Comment:
```suggestion
    track(model: SimpleModel): void {
```

SimpleModel is used until the core (player) Model module is loaded. We don't persist media (playlist item) state.